default_build_params

```diff
@@ -1,3 +1,5 @@
+#!/bin/sh
```

Comment:
This file is actually not a script, it's being sourced by other scripts, it's not even executable.  I, on purpose, did not include the hashbang line so please remove it.  But you are welcomed to add a comment to clarify the intention of the file.

Reply:
even when sourcing, that executes the script with the active shell (instead of sub-shell)
ifs and echos wouldn't work otherwise... 
only thing blocking execution is the x flag

having the shebang helps editing the script as every linter including even vim will detect it and highlight the structure

Reply:
> ifs and echos wouldn't work otherwise...

What?  I am not following here?  `if` and `echo` are working fine without the shebang line.

> having the shebang helps editing the script as every linter including even vim will detect it and highlight the structure

I can take that as an argument.  Indeed it does help for editing.

Reply:
I didn't mean that `if`/`echo` are affected by the shebang. What I mean is if the script was not being executed as if called with `[ba|c|z]sh <script>`, it would raise an error when reaching those commands. So `source` interprets the script in the same way as the shell. Calling with the shell only runs it separately instead of within the same shell.

Not adding the shebang also doesn't stop anyone to call `[ba|c|z]sh <script>`. So really the only thing the shebang does in this case is provide an indication on how to interpret the contents, and this is even more useful when there isn't an explicit `.sh` extension.

Reply:
> Not adding the shebang also doesn't stop anyone to call `[ba|c|z]sh <script>`.

When called as `[ba|c|z]sh <script>`, you are specifying the shell already so not sure how the shebang line will make the difference.  The shebang line is for the case you call the script directly as `./<script>` (without specifying the shell) but then that's not possible right now since the script is not executable.

Reply:
Yes, that's why in this case it only help formatting and readability of the file.